### ISSUE TITLE
Custom scenario-defined terrain types

### ISSUE THREAD

**Issue by kostmo** [swarm-game/swarm#1641]
Currently, all terrain types are hard coded.  There may be a use case for definitions in YAML and per-scenario extensibility as exists for Entities.

**Comment by byorgey** [comment]
Agreed, I created the existing terrain types in a very offhand sort of way and always kind of intended that there should be more, and customizable, terrain types in the future.
